lib/http.js

```diff
@@ -6,9 +6,43 @@ const EventEmitter = require('events').EventEmitter;
 
 exports.IncomingMessage = require('_http_incoming').IncomingMessage;
 
-
-const common = require('_http_common');
-exports.METHODS = common.methods.slice().sort();
+// Request Methods
+var METHODS = [
+  'DELETE',
+  'GET',
+  'HEAD',
+  'POST',
+  'PUT',
+  // pathological
+  'CONNECT',
+  'OPTIONS',
+  'TRACE',
+  // webdav
+  'COPY',
+  'LOCK',
+  'MKCOL',
+  'MOVE',
+  'PROPFIND',
+  'PROPPATCH',
+  'SEARCH',
+  'UNLOCK',
+  // subversion
+  'REPORT',
+  'MKACTIVITY',
+  'CHECKOUT',
+  'MERGE',
+  // upnp
+  'MSEARCH',
+  'NOTIFY',
+  'SUBSCRIBE',
+  'UNSUBSCRIBE',
+  // RFC-5789
+  'PATCH',
+  'PURGE'
+];
+exports.__defineGetter__('METHODS', util.deprecate(function() {
```

Comment:
I meant using [`Object.defineProperty(exports, 'METHODS', util.deprecate(...));`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) instead, because it seems that [`__defineGetter__`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/__defineGetter__) is deprecated / on track for removal. Just thinking for the future, I don't know how v8 would handle something like that though.